test/hca_metadata_api/test.py

```diff
@@ -754,6 +778,57 @@ def test_name_substitution(self):
             self.assertEqual(expected_slash_after, with_slash_after)
 
 
+class TestSchema(AzulUnitTestCase):
+
+    def json_body(self, domain: str) -> JSON:
+        return {
+            'content': {
+                'describedBy': f'https://{domain}/type/protocol/7.1.0/protocol',
+                'protocol_core': {
+                    'protocol_id': 'test-protocol'
+                },
+                'schema_type': 'protocol'
+            },
+            'hca_ingest': {
+                'document_id': str(uuid4()),
+                'submissionDate': '2024-01-01T00:00:01.000Z',
+            }
+        }
+
+    def test_valid_schema_domain(self):
+        json_body = self.json_body(domain='schema.humancellatlas.org')
+        file = Protocol.from_json(json_body)
+        self.assertIsInstance(file, Protocol)
+
+    def test_invalid_schema_domain(self):
+        json_body = self.json_body(domain='schema.staging.data.humancellatlas.org')
+        with self.assertRaises(RequirementError):
+            Protocol.from_json(json_body)
+
+    @valid_schema_domains(['schema.staging.data.humancellatlas.org'])
+    def test_patched_schema_domain(self):
+        json_body = self.json_body(domain='schema.staging.data.humancellatlas.org')
+        file = Protocol.from_json(json_body)
+        self.assertIsInstance(file, Protocol)
+
+    @valid_schema_domains(['schema.staging.data.humancellatlas.org'])
+    def test_invalid_patched_schema_domain(self):
+        json_body = self.json_body(domain='schema.humancellatlas.org')
+        with self.assertRaises(RequirementError):
+            Protocol.from_json(json_body)
+
```

Comment:
Since you are actively working on #6666, let's not have a FIXME and just add this case in that PR.